drivers/pinctrl/Kconfig.rcar

```diff
@@ -7,3 +7,11 @@ config PINCTRL_RCAR_PFC
 	depends on DT_HAS_RENESAS_RCAR_PFC_ENABLED
 	help
 	  Enable pin controller driver for Renesas RCar SoC
+
+if PINCTRL_RCAR_PFC
+
+config PINCTRL_RCAR_VOLTAGE_CONTROL
+	bool "Voltage control functionality of Renesas R-Car PFC driver"
+	default y if SOC_SERIES_RCAR_GEN3
+
+endif
```

Comment:
use depends on for single if/endif item. question: what is the reason to guard this feature?

Reply:
I believe that adding a `PINCTRL_RCAR_PFC` check here is unnecessary since the file where the configuration is used is compiled exclusively for `PINCTRL_RCAR_PFC`. So, I'll remove the checking of `PINCTRL_RCAR_PFC` from this file.

If you are inquiring about the reason behind the need for the `PINCTRL_RCAR_VOLTAGE_CONTROL` config, a partial answer can be found here: https://github.com/zephyrproject-rtos/zephyr/pull/60444#issuecomment-1662114058. The R-Car PFC driver is utilized for both Gen4 and Gen3 platforms, and currently, the voltage control table has been added only for Gen3.

While it is possible to add a check, such as `SOC_SERIES_RCAR_GEN3`, to the driver code, but as for me, using an additional config provides a clearer approach.